lib/insights/api/common/paginated_response_v2.rb

```diff
@@ -2,35 +2,122 @@ module Insights
   module API
     module Common
       class PaginatedResponseV2 < PaginatedResponse
+
+        # GraphQL name regex: /[_A-Za-z][_0-9A-Za-z]*/
+        ASSOCIATION_COUNT_ATTR = "__count".freeze
+
         attr_reader :limit, :offset, :sort_by
 
+        def records
+          @records ||= begin
+            res = @base_query.order(:id).limit(limit).offset(offset)
+
+            select_for_associations, group_by_associations = sort_by_associations_query_parameters
+            res = res.select(*select_for_associations)          if select_for_associations.present?
+            res = res.left_outer_joins(*sort_by_associations)   if sort_by_associations.present?
+            res = res.group(group_by_associations)              if group_by_associations.present?
+
+            order_options = sort_by_options(res.klass)
+            res = res.reorder(order_options) if order_options.present?
+            res
+          end
+        end
+
+        # Condenses parameter values for handling multi-level associations
+        # and returns an array of key, value pairs.
+        #
+        # Input:  { "association" => { "attribute" => "value" }, "direct_attribute" = "value2" }
+        # Output: { "association.attribute" => "value", "direct_attribute" => "value2" }
+        #
+        # Input:  { "association" => { "attribute" => "value" }, "association2" => "attribute2" = "value2" }
+        # Output: { "association.attribute" => "value", "association2.attribute2" => "value2" }
```

Comment:
From the looks of it, this will return an Array of Arrays, not a hash.  Can you turn these comments into unit tests for this method?

Reply:
Also, it doesn't look like it would cover `{"association" => {"attribute" => "value", "attribute2" = "value2"}`

Reply:
thanks for finding this.  fixing this exposed an issue when referencing same named attribute from class and association class.   will be pushing an update shortly. 👍